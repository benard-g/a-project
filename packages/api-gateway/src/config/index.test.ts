import { loadConfig } from './index';

describe('config', () => {
  let envSave: NodeJS.ProcessEnv;

  beforeEach(() => {
    envSave = process.env;
    process.env = {};
  });

  afterEach(() => {
    process.env = envSave;
  });

  describe('development', () => {
    it('should use default values', () => {
      process.env.NODE_ENV = 'development';

      const config = loadConfig();

      expect(config).toMatchSnapshot();
    });
  });

  describe('test', () => {
    it('should use default values', () => {
      process.env.NODE_ENV = 'test';

      const config = loadConfig();

      expect(config).toMatchSnapshot();
    });
  });

  describe('production', () => {
    it('should use the values from the environment', () => {
      process.env = {
        NODE_ENV: 'production',
        ACCESS_TOKEN_DURATION: '123456',
        DATABASE_URL: 'some-database-url',
        JWT_SECRET_KEY: 'some-jwt-secret',
        PORT: '4242',
      };

      const config = loadConfig();

      expect(config).toBeDefined();
    });

    it('should throw if ACCESS_TOKEN_DURATION is missing', () => {
      process.env = {
        NODE_ENV: 'production',
        DATABASE_URL: 'some-database-url',
        JWT_SECRET_KEY: 'some-jwt-secret',
        PORT: '4242',
      };

      expect(() => {
        loadConfig();
      }).toThrowError(
        'EnvironmentVariableError: Missing environment variable "ACCESS_TOKEN_DURATION"',
      );
    });

    it('should throw if DATABASE_URL is missing', () => {
      process.env = {
        NODE_ENV: 'production',
        ACCESS_TOKEN_DURATION: '123456',
        JWT_SECRET_KEY: 'some-jwt-secret',
        PORT: '4242',
      };

      expect(() => {
        loadConfig();
      }).toThrowError(
        'EnvironmentVariableError: Missing environment variable "DATABASE_URL"',
      );
    });

    it('should throw if JWT_SECRET_KEY is missing', () => {
      process.env = {
        NODE_ENV: 'production',
        ACCESS_TOKEN_DURATION: '123456',
        DATABASE_URL: 'some-database-url',
        PORT: '4242',
      };

      expect(() => {
        loadConfig();
      }).toThrowError(
        'EnvironmentVariableError: Missing environment variable "JWT_SECRET_KEY"',
      );
    });

    it('should throw if PORT is missing', () => {
      process.env = {
        NODE_ENV: 'production',
        ACCESS_TOKEN_DURATION: '123456',
        DATABASE_URL: 'some-database-url',
        JWT_SECRET_KEY: 'some-jwt-secret',
      };

      expect(() => {
        loadConfig();
      }).toThrowError(
        'EnvironmentVariableError: Missing environment variable "PORT"',
      );
    });
  });
});
