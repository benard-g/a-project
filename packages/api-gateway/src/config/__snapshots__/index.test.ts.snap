// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`config development should use default values 1`] = `
Object {
  "ACCESS_TOKEN_DURATION": 604800,
  "DATABASE_URL": "postgres://postgres:password@localhost:5432/postgres",
  "GRAPHQL_SCHEMA_GENERATION_PATH": "../../schema.graphql",
  "JWT_SECRET_KEY": "jwt-secret-key",
  "LOG_LEVEL": "debug",
  "NODE_ENV": "development",
  "PORT": 8080,
}
`;

exports[`config test should use default values 1`] = `
Object {
  "ACCESS_TOKEN_DURATION": 604800,
  "DATABASE_URL": "postgres://postgres:password@localhost:2345/postgres",
  "GRAPHQL_SCHEMA_GENERATION_PATH": false,
  "JWT_SECRET_KEY": "jwt-secret-key",
  "LOG_LEVEL": "debug",
  "NODE_ENV": "test",
  "PORT": 8080,
}
`;
